core/src/main/java/org/bitcoinj/evolution/QuorumState.java

```diff
@@ -146,6 +144,10 @@ public void applyDiff(Peer peer, AbstractBlockChain headersChain, AbstractBlockC
             quorumList.syncWithMasternodeList(newMNList);
         }
         if (peer != null && isSyncingHeadersFirst) peer.queueMasternodeListDownloadedListeners(MasternodeListDownloadedListener.Stage.ProcessedQuorums, mnlistdiff);
+
+        // save the current state, if both mnLists and quorums are both applied
+        mnListsCache.put(newMNList.getBlockHash(), newMNList);
+        mnList = newMNList;
         quorumsCache.put(newQuorumList.getBlockHash(), newQuorumList);
         quorumList = newQuorumList;
```

Comment:
found a bug where `mnList` and `quorumList` were at different heights on the blockchain, which might be because the were not "saved" at the same time.  Perhaps an exception with new quorums caused this.  

When these two lists are out of alignment, then it is not possibly to successfully update them without starting over from the beginning of the blockchain or the bootstrap (if there is one).

Reply:
An opportunity to write a test perhaps? :)

Reply:
Ideally, but it won't be easy.